boat-scaffold/src/main/templates/boat-java/libraries/resttemplate/ApiClient.mustache

```diff
@@ -96,42 +108,60 @@ public class ApiClient {
 
     private String basePath = "{{basePath}}";
 
-    private RestTemplate restTemplate;
+    private final RestTemplate restTemplate;
 
+{{#authMethods}}
     private Map<String, Authentication> authentications;
+{{/authMethods}}
 
+{{^useJacksonConversion}}
     private DateFormat dateFormat;
+{{/useJacksonConversion}}
 
+{{^serviceId}}
     public ApiClient() {
         this.restTemplate = buildRestTemplate();
         init();
     }
+{{/serviceId}}
+{{#useJacksonConversion}}
+    @Autowired
+    private ObjectMapper objectMapper;
+{{/useJacksonConversion}}
 
     @Autowired
-    public ApiClient(RestTemplate restTemplate) {
+    public ApiClient({{#serviceId}}@Qualifier(INTER_SERVICE_REST_TEMPLATE_BEAN_NAME) {{/serviceId}}RestTemplate restTemplate) {
```

Comment:
Too specific. You can externalise qualifier to a variable to so devs can choose which rest template to inject.

Reply:
The qualifier is now an option